### ISSUE TITLE
.NET MAUI SampleRate option crashes on Android.

### ISSUE THREAD

**Issue by Viktor246** [getsentry/sentry-dotnet#2997]
### Package

Sentry.Maui

### .NET Flavor

.NET Core

### .NET Version

8.0.0

### OS

Android

### SDK Version

3.41.3

### Steps to Reproduce

1. Create .NET MAUI Blazor Hybrid applicaiton using Visual Studio template.
2. Add Sentry.Maui nuget package
3. Add .UseSentry() option (code below)
4. Run app on Android emulator

```csharp
var builder = MauiApp.CreateBuilder();
builder
    .UseMauiApp<App>()
    .ConfigureFonts(fonts =>
    {
        fonts.AddFont("OpenSans-Regular.ttf", "OpenSansRegular");
    }
    ).UseSentry(o =>
    {
        o.Dsn = "xxx";
        o.SampleRate = (float?)1.0;
    });
    // ...
    return builder.Build(); // Exception is thrown on this line
```

### Expected Result

Sentry should work.

### Actual Result

InvalidCastException is thrown on line

```csharp
return builder.Build();
```
System.InvalidCastException: 'Specified cast is not valid.'

Base exception stack trace is

```
   at Sentry.SentrySdk.<>c__DisplayClass6_0.<InitSentryAndroidSdk>b__0(SentryAndroidOptions o)
   at Sentry.Android.Callbacks.OptionsConfigurationCallback.Configure(Object optionsObject)
   at Sentry.JavaSdk.Sentry.IOptionsConfigurationInvoker.n_Configure_Lio_sentry_SentryOptions_(IntPtr jnienv, IntPtr native__this, IntPtr native_p0)
   at Android.Runtime.JNINativeWrapper.Wrap_JniMarshal_PPL_V(_JniMarshal_PPL_V callback, IntPtr jnienv, IntPtr klazz, IntPtr p0) in /Users/runner/work/1/s/xamarin-android/src/Mono.Android/Android.Runtime/JNINativeWrapper.g.cs:line 125
   ```
  
![SentryError](https://github.com/getsentry/sentry-dotnet/assets/35272429/4b6c97ba-7bd8-41b7-ac46-8a12d1fea703)

When runing on windows machine everything works fine, on android emulator it crashes.

**Comment by bitsandfoxes** [comment]
Thanks for the detailed context and repro! I'll try to get this running locally as well.

**Comment by bitsandfoxes** [comment]
That one is a brainscratcher. I can irregularly reproduce this. But only in `debug`. The app starts just fine when simply launching on the device.

**Comment by IsmailHassani** [comment]
I have this issue also on the android emulator.
A couple weeks ago i came along an issue where http://localhost is not available on the emulator and http://10.0.2.2 should be used instead. I don't know where i saw this, but somewhere this sdk uses localhost for sampling.
This could explain why on debug this fails.

**Comment by bitsandfoxes** [comment]
Thanks for sharing those insights!
